### PR TITLE
[onert] Release constant tensor when ruy kernel runs on cpu

### PR DESCRIPTION
**Pull request by YongseopKim**
Release constant tensor(weight) after ruy kernel runs on cpu if possible
- if batch size <= 4, ruy kernel caches the weight so that we can
release the weight.

ONE-DCO-1.0-Signed-off-by: Yongseop Kim <yons.kim@samsung.com>

### CODE REVIEW

File: runtime/onert/backend/cpu/ops/FullyConnectedLayer.cc [hunk 1]
```diff
@@ -100,6 +101,44 @@ void FullyConnectedLayer::fullyConnectedHybrid()
...
+
+  // TODO Remove const_cast
+  auto weight_tensor = dynamic_cast<const Tensor *>(_weights);
+  if (weight_tensor)
+  {
+    // TODO Remove this workaround
+    // buffer will have deallocated memory address because ruy kernel uses
+    // it as cache key
+    auto tensor = const_cast<Tensor *>(weight_tensor);
+    auto buffer = tensor->buffer();
```
Comment (wateret) [inline runtime/onert/backend/cpu/ops/FullyConnectedLayer.cc:133]: To remove the workaround, can we make `buffer` to be a member of `FullyConnectedLayer`?

Comment (YongseopKim) [reply runtime/onert/backend/cpu/ops/FullyConnectedLayer.cc:133]: Oh, that's a good idea. I'll try. Thanks.

**Comment by YongseopKim** [comment]
PTAL @wateret and @periannath

File: runtime/onert/backend/cpu/ops/FullyConnectedLayer.h [hunk 2]
```diff
@@ -65,6 +67,11 @@ class FullyConnectedLayer : public ::onert::exec::IFunction
 
   ir::Activation _activation;
   std::unique_ptr<nnfw::cker::FCTempArena> _temp_arena;
+
+#ifdef USE_RUY_GEMV
+  bool _freed_weights = false;        // weights is already freed?
```
Comment (periannath) [inline runtime/onert/backend/cpu/ops/FullyConnectedLayer.h:72]: (Optional) I think `_weights_freed` or `_is_weights_freed` would be better.

Comment (YongseopKim) [reply runtime/onert/backend/cpu/ops/FullyConnectedLayer.h:72]: `_is_weights_freed` looks better, thanks.

File: runtime/onert/backend/cpu/ops/FullyConnectedLayer.cc [hunk 3]
```diff
@@ -95,11 +96,35 @@ void FullyConnectedLayer::fullyConnectedHybrid()
...
+      getTensorShape(_output), reinterpret_cast<float *>(_output->buffer()), temp_arena);
+
+  if (_cached_weights == nullptr || _is_weights_freed)
+    return;
+
+  auto weight_tensor = dynamic_cast<const Tensor *>(_weights);
+  if (weight_tensor)
+  {
+    auto tensor = const_cast<Tensor *>(weight_tensor);
+    tensor->decrease_ref();
```
Comment (wateret) [inline runtime/onert/backend/cpu/ops/FullyConnectedLayer.cc:121]: Ah, still need it as non-const due to `decrease_ref`! 😢

**Pull request merged**